pkg/machines/components/vmSnapshotsTab.jsx

```diff
@@ -0,0 +1,168 @@
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import cockpit from 'cockpit';
+import { vmId } from "../helpers.js";
+import { ListingTable } from "cockpit-components-table.jsx";
+import { Tooltip } from '@patternfly/react-core';
+import { InfoAltIcon } from '@patternfly/react-icons';
+
+import './vmSnapshotsTab.css';
+
+const _ = cockpit.gettext;
+
+function prettyTime(unixTime) {
+    const today = parseInt(new Date() / 1000 / 24 / 60 / 60);
+    const yesterday = today - 1;
+    const date = parseInt(unixTime / 24 / 60 / 60);
+
+    let time;
+    if (date == today) {
+        time = new Date(unixTime * 1000).toISOString()
+                .slice(11, 19);
+        return "Today " + time;
+    } else if (date == yesterday) {
+        time = new Date(unixTime * 1000).toISOString()
+                .slice(11, 19);
+        return "Yesterday " + time;
+    } else {
+        const options = { year: 'numeric', month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' };
+        return new Date(unixTime * 1000).toLocaleDateString('en-GB', options);
+    }
+}
+
+class VmSnapshotsTab extends React.Component {
+    render() {
+        const { vm } = this.props;
+        const id = vmId(vm.name);
+
+        const emptyCaption = (<div className="no-snapshots">
+            <h3>{_("No snapshots")}</h3>
+            {_("Snapshots allow you to revert to an earlier state if something goes wrong")}
+        </div>);
+
+        let detailMap = [
+            {
+                name: _("Time"), value: (snap, snapId) => {
+                    const date = prettyTime(snap.creationTime);
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-date`}>
+                            {date}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Name"), value: (snap, snapId) => {
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-name`}>
+                            {snap.name}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Description"), value: (snap, snapId) => {
+                    let desc = snap.description;
+                    if (!desc)
+                        desc = (<span className="snap-greyed-out">{_("No description")}</span>);
+
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-description`}>
+                            {desc}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("State"), value: (snap, snapId) => {
+                    const infoTips = {
+                        shutdown: _("VM is shutting down"),
```

Comment:
I think `shutting down` would work better here. Otherwise it's easy to read as "is shut down"

Reply:
or even _shutting off_, since it will lead to the state _shut off_?

Reply:
if `VM is shutting down` can easily be read as `VM is shut down`, then `shutting down` or `shutting off` can easily be read as `shut off`, since in all cases difference is only in the same 3 letters `ing`, no?

Reply:
Ah, right. We can make it `Shutting down` and `Shut off` then.